### PR TITLE
Sockets - Bita and Maria 

### PR DESCRIPTION
**Pull request by MariaWissler**
# Adagrams
Congratulations! You're submitting your assignment.

## Comprehension Questions

Feature | Feedback
--- | ---
What are the components that make up a method? | Signature and body
What are the advantages of using git when collaboratively working on one code base? | Work in progress
What kind of relationship did you and your pair have with the unit tests? | 
Does your code use any methods from the `Enumerable` mixin? If so, where and why was it helpful? | yes , it was interesting to see some of them were able to mix, specially for the scoring 
What was one method you and your pair used to debug code? | went to the code line by line, connecting what was happening in the code, and what methods were returning
What are two discussion points that you and your pair discussed when giving/receiving feedback from each other that you would be willing to share? | Work in Progress

### CODE REVIEW

File: adagrams.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# wirte a method with no parameters
```
Comment (kaidamasaki) [inline adagrams.rb:1]: In the future make sure to clean up files you aren't using.

File: lib/adagrams.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  letters_in_hand
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  input.each_char do |letter|
+    index = letters_in_hand.index(letter.upcase)
+    if index.nil?
+      return false
+    else
+      letters_in_hand.delete_at(index)
```
Comment (kaidamasaki) [inline lib/adagrams.rb:24]: Because this deletes letters as it finds them if the user inputs an invalid word the letters it _does_ use will be deleted.

For example:
```
$ ruby wave-2-game.rb
Welcome to Adagrams!
Let's draw 10 letters from the letter pool...
You have drawn the letters:
N, K, E, I, W, V, R, A, A, E
Please input a word that only uses the letters from the letter bank
NAAA
You entered in a word that contains characters not in the letter bank
Please input a word that only uses the letters from the letter bank
NA
You entered in a word that contains characters not in the letter bank
Please input a word that only uses the letters from the letter bank
```

File: lib/adagrams.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+def highest_score_from(words)
+  scores = []
+  words.each do |word|
+    score = score_word(word.upcase)
+    scores << score
+  end
+
+  scores_hash = Hash[words.zip scores]
+  puts "scores_hash #{scores_hash}"
```
Comment (kaidamasaki) [inline lib/adagrams.rb:49]: In the future you should remove debugging `puts` before submitting an assignment.

Leaving the `puts` in means the user will see extra output when they try to play the game.

```
$ ruby wave-4-game.rb
Welcome to Adagrams!
Let's draw 10 letters from the letter pool...
You have drawn the letters:
L, E, M, G, E, F, O, N, D, R
Please input your submission for the longest anagram you can come up with
LEMG
Your submitted anagram scored 7 points
Should we play another round?
Enter y to replay
y
Let's draw 10 letters from the letter pool...
You have drawn the letters:
O, P, Z, E, E, O, D, J, H, T
Please input your submission for the longest anagram you can come up with
OPZEE
Your submitted anagram scored 16 points
Should we play another round?
Enter y to replay

scores_hash {"LEMG"=>7, "OPZEE"=>16}
winner_score {"OPZEE"=>16}
winner_score.length 1
Thanks for playing Adagrams!
The highest score from this game was OPZEE, which was worth 16 points
Goodbye!
```

File: lib/adagrams.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  winner_score = scores_hash.select { |_key, value| value == scores_hash.values.max }
+  puts "winner_score #{winner_score}"
+  puts "winner_score.length #{winner_score.length}"
```
Comment (kaidamasaki) [inline lib/adagrams.rb:52]: Remove debugging `puts`.

File: lib/adagrams.rb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    end
+  end
+
+  winner = ""
+  if long_words.length == 1
+    winner = long_words[0]
+  elsif long_words.length > 1
+    winner = long_words[0]
+  else
+    short_words = winner_score.min_by { |key, _value| key.length }
```
Comment (kaidamasaki) [inline lib/adagrams.rb:67]: Try to use more descriptive names.  For example in this case you could do something like:
```ruby
short_words = winner_score.min_by { |word, _score| word.length }
```

**Comment by kaidamasaki** [comment]
# Adagrams
## What We're Looking For

Feature                                                          | Feedback
---------------------------------------------------------------- | --------
**General**                                                      |
Answered comprehension questions                                 | Partial
Small commits with meaningful commit messages                    | Small commits
**Code Requirements**                                            |
`draw_letters` method                                            | ✅
Uses appropriate data structure to store the letter distribution | ✅
All tests for `draw_letters` pass                                | ✅
`uses_available_letters?` method                                 | ✅
All tests for `uses_available_letters?` pass                     | ✅
`score_word` method                                              | ✅
Uses appropriate data structure to store the letter scores       | ✅
All tests for `score_word` pass                                  | ✅
`highest_score_from` method                                      | ✅
Appropriately handles edge cases for tie-breaking logic          | ✅
All tests for `highest_score_from` pass                          | ✅
**Overall**                                                      | Good job!  Aside from one minor bug (that wasn't in the tests) everything worked perfectly.  :)  I would like to see more descriptive commit messages. Your commits should tell the reader a story about what changed when and why, without them having to dig through the code or project requirements.
